test.js

```diff
@@ -717,4 +728,44 @@ function runTest (client, db) {
       })
     })
   })
+
+  test('subscription should expire after client disconnected', function (t) {
+    dbopts.ttl = {
+      subscriptions: 1
+    }
+    dbopts.ttlAfterDisconnected = true
+    const instance = persistence(dbopts)
+    const emitter = mqemitterMongo(dbopts)
+    const client = { id: 'client1' }
+    instance.broker = toBroker('1', emitter)
+    instance.on('ready', () => {
+      instance.addSubscriptions(client, [{
+        topic: 'hello',
+        qos: 1
+      }], function (err) {
+        t.error(err)
+        db.collection('subscriptions').findOne({ clientId: client.id, topic: 'hello' }, function (err, result) {
+          t.error(err)
+          t.notEqual(result, null, 'must return subscription')
+          t.equal(result.disconnected, undefined, 'disconnected should not be set')
+          instance.broker.emit('clientDisconnect', client)
+          setTimeout(() => {
+            db.collection('subscriptions').findOne({ clientId: client.id, topic: 'hello' }, function (err, result) {
+              t.error(err)
+              t.notEqual(result, null, 'must return subscription')
+              t.notEqual(result.disconnected, undefined, 'disconnected should be set')
+              setTimeout(() => {
+                db.collection('subscriptions').findOne({ clientId: client.id, topic: 'hello' }, function (err, result) {
+                  t.error(err)
+                  t.equal(result, null, 'must not return subscription')
+                  instance.destroy(t.pass.bind(t, 'instance dies'))
+                  emitter.close(t.end.bind(t))
+                })
+              }, 3000)
```

Comment:
And set this to 1500, just to speed up tests a bit

Reply:
That would make it fail occasionally, it can take up to 3 seconds until subscription is removed (1s ttl + 2s ttlMonitorSleepSecs), with current solution we wait 3500ms, can lower it to 2600 (3100ms total) and that way we should still have enough safety margin.

Reply:
Ok so keep it as it is